web/src/compose_actions.js

```diff
@@ -70,6 +71,23 @@ function show_compose_box(msg_type, opts) {
     // When changing this, edit the 42px in _maybe_autoscroll
     $(".new_message_textarea").css("min-height", "3em");
     compose_ui.set_focus(msg_type, opts);
+    const filter = narrow_state.filter();
+    if (filter && filter.is_search()) {
+        show_compose_in_search_view_banner();
+    }
+}
+
+function show_compose_in_search_view_banner() {
+    const search_view_banner = render_compose_banner({
+        banner_type: compose_banner.WARNING,
+        classname: compose_banner.CLASSNAMES.compose_in_search_view,
+        banner_text: $t({
+            defaultMessage:
+                "You are composing a message from a search view, which may not include the latest messages in the conversation.",
```

Comment:
maybe this should be more like "This view may not contain all messages in visible conversations."